src/machine/machine_rp2040_rng.go

```diff
@@ -0,0 +1,38 @@
+//go:build rp2040
+// +build rp2040
+
+// Implementation based on code located here:
+// https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_lwip/random.c
+
+package machine
+
+import (
+	"device/rp"
+)
+
+const numberOfCycles = 32
+
+// GetRNG returns 32 bits of semi-random data based on ring oscillator.
+func GetRNG() (uint32, error) {
+	var val uint32
+	for i := 0; i < 4; i++ {
+		val = (val << 8) | uint32(roscRandByte())
+	}
+	return val, nil
+}
+
+var val uint8
```

Comment:
Given this is now a global in the `machine` package, does it deserve a better name?

Reply:
Renamed.